docs/book/src/reference/writing-tests.md

```diff
@@ -1,19 +1,33 @@
 # Writing controller tests
 
-Testing Kubernetes controller is a big subject, and the boilerplate testing
+Testing Kubernetes controllers is a big subject, and the boilerplate testing
 files generated for you by kubebuilder are fairly minimal.
 
-[Writing and Running Integration Tests](/reference/testing/envtest.md) documents steps to consider when writing integration steps for your controllers, and available options for configuring your test control plane using [`envtest`](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/envtest).
+To walk you through integration testing patterns for Kubebuilder-generated controllers, we will revisit the CronJob we built in our first tutorial and write a simple test for it.
 
-Until more documentation has been written, your best bet to get started is to look at some
-existing examples, such as:
+The basic approach is that, in your generated `suite_test.go` file, you will create a local Kubernetes API server, instantiate and run your controllers, and then write additional `*_test.go` files to test it using [Ginko](http://onsi.github.io/ginkgo).
+
+## Test Environment Setup
+
+{{#literatego ../cronjob-tutorial/testdata/project/controllers/suite_test.go}}
+
+## Testing your Controller's Behavior
+
+{{#literatego ../cronjob-tutorial/testdata/project/controllers/cronjob_controller_test.go}}
+
+This Status update example above demonstrates a general testing strategy for a custom Kind with downstream objects. By this point, you hopefully have learned the following methods for testing your controller behavior:
+
+* Setting up your controller to run on an envtest cluster
+* Writing stubs for creating test objects
+* Isolating changes to an object to test specific controller behavior.
+
+If you want to tinker with how your envtest cluster is configured, see section [Writing and Running Integration Tests](/reference/testing/envtest.md) as well as the [`envtest docs`](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/envtest).
```

Comment:
This is probably a bit subjective but I sort of liked how the current doc points me to the [Writing and Running Integration Tests](https://book.kubebuilder.io/reference/testing/envtest.html) so I can get a bit of an overview of envtest, and the configurations and caveats to consider before jumping into an example.

So I'm wondering if we want to move these lines to the top of the doc as part of the overview to better highlight this. Currently I have to scroll quite a bit through the example to learn more about envtest itself.

Reply:
I'm happy to move these lines to the top of the doc. I agree, a lot of users might want to have this envtest information up front.